torch/onnx/_internal/fx/export.py

```diff
@@ -500,6 +559,184 @@ def _validate_op_between_ort_torch(
         warnings.warn(f"ORT fails to run with error: {e}.")
 
 
+def _location_from_fx_stack_trace(
+    node_stack_trace: str,
+) -> Optional[diagnostics.infra.Location]:
+    """Extract location from FX node stack trace.
+
+    Args:
+        node_stack_trace: The stack trace of the FX node. Example:
+
+            File "path/file.py", line 311, in <function>
+                <code>
+            |   File "path/file2.py", line 389, in <function>
+                <code>
+
+    Returns:
+        location: The location of the FX node.
+    """
+    lines = node_stack_trace.strip().split("\n")
+    idx = 0
+    while idx < len(lines) and "File" not in lines[idx]:
+        idx += 1
+    if idx + 1 >= len(lines):
+        return None
+
+    pattern = re.compile(r"^File \"(.+)\", line (\d+), in (.+)$")
+    matches = pattern.match(lines[idx].strip())
+    if matches:
+        uri = matches.group(1)
+        line_number = int(matches.group(2))
+        snippet = lines[idx + 1].strip()
+        return diagnostics.infra.Location(uri=uri, line=line_number, snippet=snippet)
+    return None
+
+
+@diagnostics.diagnose_call(
```

Comment:
Does it cover the whole diagnostic? For example, do we need another decorator for `_validate_op_between_ort_torch`, or anything new we add into the exporter?

Reply:
good point, adding a few lines to `_validate_op_between_ort_torch` to log the message.

Final result looks like this
![image](https://user-images.githubusercontent.com/9376104/217679451-5dfc4e38-ae92-4d31-a628-d8b80d26e9f2.png)